src/uct/ib/dc/dc_mlx5.c

```diff
@@ -242,7 +242,7 @@ static unsigned uct_dc_mlx5_iface_progress(void *arg)
     unsigned count;
 
     count = uct_rc_mlx5_iface_common_poll_rx(&iface->super, 0);
-    if (count > 0) {
+    if ((count > 0) && !iface->super.super.config.tx_poll_always) {
```

Comment:
maybe as experiment for perf - instead of adding this cond here, return "!iface->super.super.config.tx_poll_always" instead of "count" from uct_rc_mlx5_iface_common_poll_rx() when uct_rc_mlx5_iface_poll_rx_cq() is non-NULL, so the rx_poll function will return 0, forcing a tx poll call.

Reply:
i'll give it a try